File: Snake Procession/Snake Procession.c
#include <stdio.h>
#include<string.h>

int main(void) {
	// your code goes here
	int t;
	scanf("%d",&t);
	while(t--)
	{
	   int n,game=0,i=0;
	   scanf("%d",&n);
	   char a[n+1];
	   scanf("%s",a);
	   for( i =0; i<strlen(a); i++)
	   {
	      if(a[i] == 'H')
	      {
	         game++;
	      }
	      if(a[i] == 'T')
	      {
	         game--;
	      }
	      if(game>1)
	      {
	         printf("Invalid\n");
	         break;
	      }
	      if(game<0)
	      {
	         printf("Invalid\n");
	         break;
	      }
	   }
	   if(game!=0 && i == strlen(a))
	   {
	      printf("Invalid\n");
	   }
	   else if (i == strlen(a))
	   {
	      printf("Valid\n");
	   }
	}
	return 0;
}

File: Snake Procession/Snake Procession.py
# cook your dish here
import math
t = int(input())
 
while t:
    n = int(input())
    l = list(filter((".").__ne__, input()))  
    if len(l)== 0:
        print("Valid")
    elif len(l)%2 !=0 :
        print("Invalid")
    else:
        validity = "Valid"
        i=0
        while i< len(l)-1:
            if (l[i]== "H" and l[i+1] != "T") or l[i]== "T":
                validity = "Invalid"
            i+=2
        print(validity)
    t-=1
